mkosi

```diff
@@ -3745,6 +3759,10 @@ def check_root():
     if os.getuid() != 0:
         die("Must be invoked as root.")
 
+def check_native(args: CommandLineArguments) -> None:
+    if args.arch is not None and args.arch != platform.machine() and args.build_script:
+        die('Cannot (currently) override the architecture and run build commands')
```

Comment:
We could eventually make this useful for https://github.com/systemd/mkosi/issues/138#issuecomment-338143991

Reply:
Right now, when I call dnf for archs others than amd64  and i386, installation fails. But they might work if binfmt is set up properly. I need to figure this out. If that is true, then #138 would just work.

Reply:
@keszybz This is documented here as a way that works (at least for me): https://wiki.mageia.org/en/Using_DNF#Setting_up_a_container_for_a_non-native_architecture